src/main/java/org/opensearch/neuralsearch/search/query/HybridQueryPhaseSearcher.java

```diff
@@ -55,17 +62,127 @@ public HybridQueryPhaseSearcher() {
         final boolean hasFilterCollector,
         final boolean hasTimeout
     ) throws IOException {
-        if (query instanceof HybridQuery) {
+        if (isHybridQuery(query, searchContext)) {
+            query = extractHybridQuery(searchContext, query);
             return searchWithCollector(searchContext, searcher, query, collectors, hasFilterCollector, hasTimeout);
         }
+        validateQuery(searchContext, query);
         return super.searchWith(searchContext, searcher, query, collectors, hasFilterCollector, hasTimeout);
     }
 
+    private boolean isHybridQuery(final Query query, final SearchContext searchContext) {
+        if (query instanceof HybridQuery) {
+            return true;
+        } else if (hasNestedFieldOrNestedDocs(query, searchContext) && mightBeWrappedHybridQuery(query)) {
+            // checking if this is a hybrid query that is wrapped into a Bool query by core Opensearch code
+            // https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/DefaultSearchContext.java#L367-L370.
+            // main reason for that is performance optimization, at time of writing we are ok with loosing on performance if that's unblocks
+            // hybrid query for indexes with nested field types.
+            // in such case we consider query a valid hybrid query. Later in the code we will extract it and execute as a main query for
+            // this search request.
+            // below is sample structure of such query:
+            //
+            // Boolean {
+            // should: {
+            // hybrid: {
+            // sub_query1 {}
+            // sub_query2 {}
+            // }
+            // }
+            // filter: {
+            // exists: {
+            // field: "_primary_term"
+            // }
+            // }
+            // }
+            // TODO Need to add logic for passing hybrid sub-queries through the same logic in core to ensure there is no latency regression
+            if (query instanceof BooleanQuery == false) {
+                return false;
+            }
+            return ((BooleanQuery) query).clauses()
+                .stream()
+                .filter(clause -> clause.getQuery() instanceof HybridQuery == false)
+                .allMatch(clause -> {
+                    return clause.getOccur() == BooleanClause.Occur.FILTER
+                        && clause.getQuery() instanceof FieldExistsQuery
+                        && SeqNoFieldMapper.PRIMARY_TERM_NAME.equals(((FieldExistsQuery) clause.getQuery()).getField());
+                });
+        }
+        return false;
+    }
+
+    private boolean hasNestedFieldOrNestedDocs(final Query query, final SearchContext searchContext) {
+        return searchContext.mapperService().hasNested() && new NestedHelper(searchContext.mapperService()).mightMatchNestedDocs(query);
+    }
+
+    private boolean mightBeWrappedHybridQuery(final Query query) {
```

Comment:
Any reason why we use `might` here? Does that mean, there is a case where it is not wrapped hybrid query even if this method return true? 
```suggestion
    private boolean isWrappedHybridQuery(final Query query) {
```

Reply:
ack, let me rename method as suggested